fearless/Common/Validation/Validators/WarningConditionViolation.swift

```diff
@@ -0,0 +1,24 @@
+import Foundation
+
+final class WarningConditionViolation: DataValidating {
+    let checkCondition: () -> Bool
+    let onWarning: (DataValidatingDelegate) -> Void
+
+    init(
+        onWarning: @escaping (DataValidatingDelegate) -> Void,
+        checkCondition: @escaping () -> Bool
+    ) {
+        self.checkCondition = checkCondition
+        self.onWarning = onWarning
+    }
+
+    func validate(notifying delegate: DataValidatingDelegate) -> DataValidationProblem? {
+        if checkCondition() {
+            return nil
+        }
+
+        onWarning(delegate)
```

Comment:
I would rewrite it to make look more natural:
```
if violatesCondition() {
    notify(delegate)
    return .warning
}
```
or maybe  
`if satisfiesCondition() {`

Reply:
Will fix in next PR